lib/quayio/scanner/image.rb

```diff
@@ -44,14 +45,22 @@ def tag
       def raw_image
         return @raw_image if defined? @raw_image
 
-        @raw_image = begin
-          JSON.parse(
-            RestClient.get("https://quay.io/api/v1/repository/#{repo}/tag/#{tag}/images",
-                           authorization: "Bearer #{quayio_token}", accept: :json)
-          )['images'].first
-        rescue RestClient::ExceptionWithResponse => err
-          return nil if err.http_code == 404 # ignore unknown repos
-          raise err
+        (1..MAX_ATTEMPTS).each do |attempt|
+          begin
+            response = RestClient.get(
+              "https://quay.io/api/v1/repository/#{repo}/tag/#{tag}/images",
+              authorization: "Bearer #{quayio_token}",
+              accept: :json)
+          rescue RestClient::ExceptionWithResponse => err
+            return nil if err.http_code == 404 # ignore unknown repos
+            if err.http_code == 520 and attempt < MAX_ATTEMPTS
```

Comment:
Warum versucht ihr es nur bei 520? Das ist ja sogar ein Spezialfall von Cloudflare. Warum nehmt ihr nicht einfach alle 500?

```suggestion
            if err.http_code >= 500 and attempt < MAX_ATTEMPTS
```

Reply:
Hm, gute Frage, ich bin unentschlossen. Uns nervt gerade der 520, weil wir wissen, dass das ein temporäres Problem ist und das weggeht, wenn zu einem späteren Zeitpunkt ein erneuter Request kommt. Andere 500 haben wir nicht gesehen und wir wissen auch nicht, wie dann die Situation aussieht. Daher würde ich für den Moment abwarten, bis wir in eine solche Situation kommen und dann bewerten, ob wir alle 500 abfangen.